python/lsst/sims/catUtils/supernovae/snObject.py

```diff
@@ -589,6 +590,94 @@ def SNObjectSED(self, time, wavelen=None, bandpass=None,
         SEDfromSNcosmo.addCCMDust(a_x=ax, b_x=bx, ebv=self.ebvofMW)
         return SEDfromSNcosmo
 
+    def SNObjectSourceSED(self, time, wavelen=None):
+        """
+        Return the rest Frame SED of SNObject at the phase corresponding to
+        time, at rest frame wavelengths wavelen. If wavelen is None,
+        then the SED is sampled at the rest frame wavelengths native to the
+        SALT model being used.
+
+        Parameters
+        ----------
+        time : float, mandatory,
+            observer frame time at which the SED has been requested in units
+            of days.
+        wavelen : `np.ndarray`, optional, defaults to native SALT wavelengths
+            array of wavelengths in the rest frame of the supernova in units
+            of nm. If None, this defaults to the wavelengths at which the
+            SALT model is sampled natively.
+        Returns
+        -------
+        `numpy.ndarray` of dtype float.
+
+        .. note: The result should usually match the SALT source spectrum.
+        However, it may be different for the following reasons:
+        1. If the time of observation is outside the model range, the values
+            have to be inserted using additional models. Here only one model
+            is currently implemented, where outside the model range the value
+            is set to 0.
+        2. If the wavelengths are beyond the range of the SALT model, the SED
+            flambda values are set to `np.nan` and these are actually set to 0.
```

Comment:
Okay.  I got confused by the '.' after '0'.  I interpreted that as an end-of-sentence period rather than a this-is-a-float decimal point.  Maybe make the '0.' into '0.0'.